CMakeLists.txt

```diff
@@ -165,9 +165,7 @@ endif()
 
 if(NOT CONFIG_NATIVE_APPLICATION)
 zephyr_ld_options(
-  -nostartfiles
-  -nodefaultlibs
-  -nostdlib
```

Comment:
Since we are touching this, might as well fix it at the same time.

"-nostdlib" is equivalent to "-nostartfiles -nodefaultlibs"

so specifying all three is redundant, we should remove -nostartfiles and -nodefaultlibs.

From the docs:

> -nostartfiles
> Do not use the standard system startup files when linking. The standard system libraries are used normally, unless -nostdlib, -nolibc, or -nodefaultlibs is used.
> 
> -nodefaultlibs
> Do not use the standard system libraries when linking. Only the libraries you specify are passed to the linker, and options specifying linkage of the system libraries, such as -static-libgcc or -shared-libgcc, are ignored. The standard startup files are used normally, unless -nostartfiles is used.
> 
> The compiler may generate calls to memcmp, memset, memcpy and memmove. These entries are usually resolved by entries in libc. These entry points should be supplied through some other mechanism when this option is specified.
> 
> -nostdlib
> Do not use the standard system startup files or libraries when linking. No startup files and only the libraries you specify are passed to the linker, and options specifying linkage of the system libraries, such as -static-libgcc or -shared-libgcc, are ignored.
> 
> The compiler may generate calls to memcmp, memset, memcpy and memmove. These entries are usually resolved by entries in libc. These entry points should be supplied through some other mechanism when this option is specified.
> 
> One of the standard libraries bypassed by -nostdlib and -nodefaultlibs is libgcc.a, a library of internal subroutines which GCC uses to overcome shortcomings of particular machines, or special needs for some languages. (See Interfacing to GCC Output in GNU Compiler Collection (GCC) Internals, for more discussion of libgcc.a.) In most cases, you need libgcc.a even when you want to avoid other standard libraries. In other words, when you specify -nostdlib or -nodefaultlibs you should usually specify -lgcc as well. This ensures that you have no unresolved references to internal GCC library subroutines. (An example of such an internal subroutine is __main, used to ensure C++ constructors are called; see collect2 in GNU Compiler Collection (GCC) Internals.)

Reply:
Coincidentally, when those two flags are dropped we no longer need the NOSYSDEF_CFLAG since xcc supports -nostdlib.

Reply:
I have uploaded a new revision with those two removed.